server/pulp/plugins/util/publish_step.py

```diff
@@ -824,6 +824,34 @@ def process_main(self):
         shutil.move(os.path.join(self.source_dir, tar_file_name), self.publish_file)
 
 
+class CreatePulpManifestStep(Step):
+    """
+    This will create a PULP_MANIFEST file in the specified directory. This step should be used when
+    the checksums of the files are not already known, because it will read and calculate new
+    checksums for each one.
+
+    If you already know the SHA256 checksums of the files going in the manifest, see an example
+    in the FileDistributor that creates this file in a different way.
+    """
+    def __init__(self, target_dir):
```

Comment:
I think that would be a great enhancement once we have something that could use it. The only other place we have that does this is in the FileDistributor, where writing the manifest is already built in to the step. It's also just 4 lines of code: open file, create writer, write, close file.

I think only the ISO distributor subclasses the FileDistributor. The use cases and context are sufficiently different that it's not worth refactoring to pull those 4 lines out into this separate step.